flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherResourceCleanupTest.java

```diff
@@ -186,6 +193,9 @@ private void startDispatcher(
                                 // because cleaning it will trigger the closeAsync latch
                                 // provided by TestingJobManagerRunner
                                 .withLocallyCleanableResource(jobManagerRunnerRegistry)
+                                .withGloballyCleanableResource(
+                                        DispatcherResourceCleanerFactory.ofLocalResource(
+                                                jobManagerRunnerRegistry))
```

Comment:
Why do we need to add this here? The `JobManagerRegistry` needs to be added explicitly because of `testJobSubmissionUnderSameJobId` where we're waiting for the JobManagerRunner to close first. But I don't see a need to add it as a global cleanup because we're not doing something similar for this.